cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/S3StorageDriver.java

```diff
@@ -172,152 +186,84 @@ public void upload(BackupRestoreContext ctx) throws IOException, URISyntaxExcept
-
-                            LOGGER.info(
-                                    "Initiating upload for file: {} | bucket: {} | key: {}",
-                                    file.getAbsolutePath(), bucketName, fileKey);
-
-                            amazonS3Client.putObject(bucketName,fileKey,file);
-
-                            LOGGER.debug(
-                                        "Successfully uploaded the file to S3. Deleting the file now: {}",
-                                        file.getAbsolutePath());
-                            final boolean delete = file.delete();
-                            LOGGER.debug("Deletion status: {} for file {}",
-                                        delete, file.getAbsolutePath());
-
-                        }
-
-                        return FileVisitResult.CONTINUE;
-                    }
-
-                    @Override
-                    public FileVisitResult visitFileFailed(Path file,
-                                                           IOException exc)
-                            throws IOException {
-                        return FileVisitResult.CONTINUE;
-                    }
-
-                    @Override
-                    public FileVisitResult postVisitDirectory(Path dir,
-                                                              IOException exc)
-                            throws IOException {
-                        return FileVisitResult.CONTINUE;
-                    }
-                });
-    }
-
     @Override
     public void download(BackupRestoreContext ctx) throws IOException, URISyntaxException {
-        // Ex: data/<keyspace>/<cf>/snapshots/</snapshot-dir>/<files>
-
-        // Location of data directory, where the data will be copied.
-        final String localLocation = ctx.getLocalLocation();
+        // download sstables at data/keyspace/cf/<files>
         final String backupName = ctx.getName();
         final String nodeId = ctx.getNodeId();
-
+        final File[] keyspaces = getNonSystemKeyspaces(ctx);
         final String bucketName = getBucketName(ctx);
+        final TransferManager tx = getS3TransferManager(ctx);
         final AmazonS3Client amazonS3Client = getAmazonS3Client(ctx);
 
-        final Map<String, Long> snapshotFileKeys = listSnapshotFiles(amazonS3Client, bucketName, backupName + "/" + nodeId);
-
-        LOGGER.info("Snapshot files for this node: {}", snapshotFileKeys);
-
-        for (String fileKey : snapshotFileKeys.keySet()) {
-            downloadFile(localLocation, bucketName, amazonS3Client, fileKey,
-                    snapshotFileKeys.get(fileKey));
+        for (File keyspace : keyspaces) {
+            for (File cfDir : keyspace.listFiles(
+                    (current, name) -> new File(current, name).isDirectory())) {
+                final String columnFamily = cfDir.getName().substring(0, cfDir.getName().indexOf("-"));
```

Comment:
Can you please move this to a helper function, as this is being use at more then 1 place. That way we can also test it.